autoload/neoterm.vim

```diff
@@ -255,9 +255,10 @@ function! s:winid()
 endfunction
 
 function! s:expand(command)
-  let l:command = substitute(a:command, '[^\\]%\(:[phtre]\)\+', '\=expand(submatch(0))', 'g')
+  let l:command = substitute(a:command, '\(\\\)\@<!%\(:[phtre]\)\+', '\=expand(submatch(0))', 'g')
   let l:command = substitute(l:command, '\c\\<cr>', g:neoterm_eof, 'g')
   let l:path = g:neoterm_use_relative_path ? expand('%') : expand('%:p')
+  let l:command = substitute(l:command, '\(\\\)\@<!%', l:path, 'g')
 
-  return substitute(l:command, '[^\\]%', l:path, 'g')
+  return substitute(l:command, '\\%', '%', 'g')
```

Comment:
the same regexp used on the line 258 (`\(\\\)\@<!%`) must be used here.

Reply:
Otherwise, we'd end with the opposit behaviour.
![nt](https://user-images.githubusercontent.com/120483/43197027-d0ed66e6-9001-11e8-9094-fbbd21696107.gif)

Reply:
What shell are you using, I tried with quotes and got the same result I got before:
![nt](https://user-images.githubusercontent.com/120483/43396668-283c8c28-93fa-11e8-8098-1e0756f61be8.gif)